tests/std/tests/GH_005244_regex_escape_sequences/test.cpp

```diff
@@ -0,0 +1,882 @@
+    g_regexTester.should_match("a1", R"(\(a\)[\1])", option);
+    g_regexTester.should_match("a\\", R"(\(a\)[\1])", option);
+    g_regexTester.should_not_match("aa", R"(\(a\)[\1])", option);
+    g_regexTester.should_match("a0", R"(\(a\)[\0])", option);
+    g_regexTester.should_match("a\\", R"(\(a\)[\0])", option);
+    g_regexTester.should_match("1a", R"([\1]\(a\))", option);
+    g_regexTester.should_match("\\a", R"([\1]\(a\))", option);
+    g_regexTester.should_not_match("aa", R"([\1]\(a\))", option);
+}
+
+void test_gh_5244_classescape_extended_or_egrep(syntax_option_type option) {
+    test_gh_5244_classescape_posix_not_awk(option);
+
+    // check no backreference handling
+    g_regexTester.should_match("a1", R"((a)[\1])", option);
+    g_regexTester.should_match("a\\", R"((a)[\1])", option);
+    g_regexTester.should_not_match("aa", R"((a)[\1])", option);
+    g_regexTester.should_match("a0", R"((a)[\0])", option);
+    g_regexTester.should_match("a\\", R"((a)[\0])", option);
+    g_regexTester.should_match("1a", R"([\1](a))", option);
+    g_regexTester.should_match("\\a", R"([\1](a))", option);
+    g_regexTester.should_not_match("aa", R"([\1](a))", option);
+}
+
+void test_gh_5244_classescape_awk() {
+    // awk-only sequences and backslash
+    g_regexTester.should_match("\\", "[\\\\]", awk);
+    g_regexTester.should_not_match("g", "[\\\\]", awk);
+    check_classescape_controlescape("\a", "a", awk);
+    check_classescape_controlescape("\b", "b", awk);
+    check_classescape_controlescape("\f", "f", awk);
+    check_classescape_controlescape("\n", "n", awk);
+    check_classescape_controlescape("\r", "r", awk);
+    check_classescape_controlescape("\t", "t", awk);
+    check_classescape_controlescape("\v", "v", awk);
+    check_classescape_identityescape("\"", awk);
+    check_classescape_identityescape("/", awk);
+
+    // awk supports octal sequences
+    g_regexTester.should_match("\1", R"([\1])", awk);
+    g_regexTester.should_not_match("1", R"([\1])", awk);
+    g_regexTester.should_not_match("\\", R"([\1])", awk);
+    g_regexTester.should_match("\11", R"([\11])", awk);
+    g_regexTester.should_not_match("1", R"([\11])", awk);
+    g_regexTester.should_not_match("\\", R"([\11])", awk);
+    g_regexTester.should_match("\111", R"([\111])", awk);
+    g_regexTester.should_not_match("0", R"([\111])", awk);
+    g_regexTester.should_not_match("\\", R"([\111])", awk);
+    g_regexTester.should_match("\111", R"([\1111])", awk);
+    g_regexTester.should_match("1", R"([\1111])", awk);
+    g_regexTester.should_not_match("\\", R"([\1111])", awk);
+    g_regexTester.should_match("A", R"([\101])", awk);
+    g_regexTester.should_not_match("\\", R"([\101])", awk);
+    g_regexTester.should_not_match("1", R"([\101])", awk);
+    g_regexTester.should_not_match("0", R"([\101])", awk);
+    g_regexTester.should_not_match("\1", R"([\101])", awk);
+    g_regexTester.should_not_match("\0", R"([\101])", awk);

```

Comment:
Plain `"\0"` will result in an empty `std::string` being constructed for the function call. We need `"\0"s`.